crates/wast/src/wast.rs

```diff
@@ -58,28 +58,36 @@ impl WastContext {
         linker.allow_shadowing(true);
         Self {
             current: None,
-            instances: HashMap::new(),
             linker,
             store,
+            modules: Vec::new(),
         }
     }
 
-    fn get_instance(&self, instance_name: Option<&str>) -> Result<Instance> {
-        match instance_name {
-            Some(name) => self
-                .instances
-                .get(name)
-                .cloned()
-                .ok_or_else(|| anyhow!("failed to find instance named `{}`", name)),
+    fn get_export(&self, module: Option<&str>, name: &str) -> Result<&Extern> {
+        match module {
+            Some(module) => {
+                let mut items = self.linker.get_by_name(module, name);
+                let ret = items
+                    .next()
+                    .ok_or_else(|| anyhow!("no item named `{}` in `{}`", name, module))?;
+                if items.next().is_some() {
+                    bail!("too many items named `{}` in `{}`", name, module);
+                }
+                return Ok(ret);
```

Comment:
Could we factor this out into a method on `Linker`, something like `self.linker.get_one_by_name(module, name)` or so? I realize it'll require a special error type to be able to report both "no item named {}" and "too many items named {}", but this seems like it'll be a common use case -- I'm even picturing something like the wasmtime CLI but backed by a Linker so that you can provide multiple modules and then do `--invoke`.